modules/markup/html_commit.go

```diff
@@ -102,6 +103,11 @@ func fullHashPatternProcessor(ctx *RenderContext, node *html.Node) {
 		if ret.QueryHash != "" {
 			text += " (" + ret.QueryHash + ")"
 		}
+		// only turn commit links to the current instance into hash link
+		if !strings.HasPrefix(strings.ToLower(ret.FullURL), strings.ToLower(setting.AppURL)) {
```

Comment:
Multidomain support #8697

Reply:
So what do we check instead of `setting.AppURL`? Is there a slice of "current app urls" somewhere?

Reply:
I guess this can be used: https://github.com/go-gitea/gitea/blob/eddf8759926911c465b249de5f6d68c052a539e0/modules/httplib/url.go#L58